src/dns_cache.rs

```diff
@@ -373,64 +373,56 @@ impl DnsCache {
     pub(crate) fn refresh_due_srv(&mut self, ty_domain: &str) -> (HashSet<String>, HashSet<u64>) {
         let now = current_time_millis();
 
-        let mut instances = vec![];
-
-        // find instance names for ty_domain.
-        for record in self.ptr.get_mut(ty_domain).into_iter().flatten() {
-            if record.get_record().is_expired(now) {
-                continue;
-            }
-
-            if let Some(ptr) = record.any().downcast_ref::<DnsPointer>() {
-                instances.push(ptr.alias.clone());
-            }
-        }
+        let instances = Self::find_instance_names(&self.ptr, ty_domain, now);
 
         // Check SRV records.
         let mut refresh_due = HashSet::new();
         let mut new_timers = HashSet::new();
         for instance in instances {
             let refresh_timers: HashSet<u64> = self
                 .srv
-                .get_mut(&instance)
+                .get_mut(instance)
                 .into_iter()
                 .flatten()
                 .filter_map(|record| record.updated_refresh_time(now))
                 .collect();
 
             if !refresh_timers.is_empty() {
-                refresh_due.insert(instance);
+                refresh_due.insert(instance.clone());
                 new_timers.extend(refresh_timers);
             }
         }
 
         (refresh_due, new_timers)
     }
 
+    fn find_instance_names<'a>(
+        from: &'a HashMap<String, Vec<DnsRecordBox>>,
+        ty_domain: &str,
+        now: u64,
+    ) -> Vec<&'a String> {
```

Comment:
Using `&self` instead of `from` would return a reference that would be tied to the entire self, and we cant do that without cloning the entire vector because we have a mutable reference here:
https://github.com/keepsimple1/mdns-sd/blob/d117f4f1b9edb425c3428750d8c69d8a2813bf78/src/dns_cache.rs#L395
So, instead, as you also mentioned, I reverted the extraction, and by such, this PR is just a perf improvement as it avoids many `.clone` calls.